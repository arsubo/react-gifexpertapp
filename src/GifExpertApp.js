import React, { useState } from 'react';

import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCatagories = [] }) => {

    const [categories, setCategories] = useState(defaultCatagories);

    // const handleAdd = () => {
    //     // setCategories(['HunterxHunter', ...categories]);
    //     setCategories(cats => [...cats, 'HunterXHunter'])
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />


            <ol>
                {
                    categories.map(category =>
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </>
    )
}

export default GifExpertApp;