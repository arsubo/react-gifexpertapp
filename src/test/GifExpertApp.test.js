import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Validaciones para el componente <GifExpertApp />', () => {
    test('Prueba que el componente <GifExpertApp/> carga bien con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Shakira'];
        const wrapper = shallow(<GifExpertApp defaultCatagories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})