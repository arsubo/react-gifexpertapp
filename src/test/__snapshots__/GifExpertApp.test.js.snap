// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validaciones para el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Shakira"
      key="Shakira"
    />
  </ol>
</Fragment>
`;

exports[`Validaciones para el componente <GifExpertApp /> Prueba que el componente <GifExpertApp/> carga bien con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
